components/aks-mis/demo.tf

```diff
@@ -0,0 +1,35 @@
+data "azurerm_user_assigned_identity" "sops_demo_mi" {
+  name                = "aks-${var.environment}-mi"
+  resource_group_name = "genesis-rg"
+}
+
+resource "azurerm_role_assignment" "demo_externaldns_read_rg" {
+  count                = (contains(["demo"], var.environment) ? 1 : 0) * var.cluster_count
+  provider             = azurerm.dts-cftptl-intsvc
+  principal_id         = data.azurerm_user_assigned_identity.sops_demo_mi.principal_id
+  scope                = "/subscriptions/1baf5470-1c3e-40d3-a6f7-74bfbce4b348/resourceGroups/core-infra-intsvc-rg"
+  role_definition_name = "Reader"
+}
```

Comment:
can you just lift the SDS code for this rather than rewriting in a different way?
https://github.com/hmcts/aks-sds-deploy/blob/3e19d0b31148743341bd6b14c6a9de2a7664592d/components/05-mis/10-managed-identity.tf#L54-L84

Reply:
Sure, I originally looked here but saw we have something similar in code for preview - https://github.com/hmcts/aks-cft-deploy/blob/main/components/aks-mis/preview.tf#L21-L48 so thought it may be best to follow that pattern in the same repo, happy to update it to this way though if preferred @timja

Reply:
only this bit is relevant https://github.com/hmcts/aks-cft-deploy/blob/main/components/aks-mis/preview.tf#L34-L48

the other bit you linked is for something else.

it would be good to bring in the sds bit as it will clean up the code. abstracting away the environments in a much nicer way

Reply:
Shall I add this to the above mentioned ticket to refactor these role assignments as well as the `targetCommand` line to match SDS, or just change it in here for now?

Reply:
https://github.com/hmcts/aks-cft-deploy/pull/273#discussion_r1004322616